Makefile

```diff
@@ -72,15 +88,22 @@ build-docker:
 build-docker-git:
 	git archive HEAD | docker build $(DOCKER_ARGS) -t $(DOCKER_TAG) -
 
-Gopkg.lock: Gopkg.toml
-	mkdir -p .go/src/github.com/zededa && ln -s ../../../.. .go/src/github.com/zededa/go-provision || :
-	GOPATH=$(CURDIR)/.go go get github.com/golang/dep/cmd/dep
-	rm vendor
-	mv src/vendor vendor
-	cd .go/src/github.com/zededa/go-provision ; PATH=$(CURDIR)/.go/bin:$$PATH GOPATH=$(CURDIR)/.go \
-          dep ensure -update $(GODEP_NAME)
-	mv vendor src/vendor
-	ln -s src/vendor vendor	
+export BUILD_CONTAINER
+eve-build-$(USER):
+	@echo "$$BUILD_CONTAINER" | docker build -t $@ - >/dev/null
```

Comment:
Is this absolutely necessary? Why can we not just build straight off of `golang:<version>` as our image? Building a custom build image seems overkill. If there _is_ a reason, then let's document it here in the `Makefile`.

Reply:
I think it is necessary because  we have to match UID/GID and USER/GROUP names between host and container. Hence there's no way we can build a universal container for that.

Why do we have to do that? Well, because of Linux users -- there Docker containers really do run as containers and UID/GID seep through. Hence all the files created by processes run within the container better be owned by the same UID/GID that is the user executing docker run. Otherwise you end up with files on the host that only root can do anything with.

Now, it could be argued that we can play the same tricks at the level of actual container execution (via sudo) but my experience tells me that it is very brittle.

Finally, there's no overhead cost (aside from the initial run) to actually build this container on the fly. Docker cache takes care of all the subsequent runs being pretty much instantaneous.

Reply:
> Well, because of Linux users

Oh yes, quite. When you run it on d4m or d4w, vpnkit takes care of a lot of this mess. Running it straight on linux means it goes, well, straight, and we should support doing it correctly as a first-class citizen.